src/app/_components/BlockList/LayoutA/Paginated.tsx

```diff
@@ -0,0 +1,118 @@
+'use client';
+
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import { Section } from '../../../../common/components/Section';
+import { ExplorerErrorBoundary } from '../../ErrorBoundary';
+import { UISingleBlock } from '../types';
+import { BlockListWithControls } from './BlockListWithControls';
+import { BlockListProvider } from './Provider';
+import { FADE_DURATION } from './consts';
+import { useBlockListContext } from './context';
+import { useBlockListWebSocket } from './useBlockListWebSocket';
+import { usePaginatedBlockList } from './usePaginatedBlockList';
+
+function PaginatedBlockListLayoutABase() {
+  const { setFadeEffect, liveUpdates } = useBlockListContext();
+  const [latestBlocksToShow, setLatestBlocksToShow] = useState<UISingleBlock[]>([]);
+
+  const { initialBlockList, initialBurnBlocks, updateList, hasNextPage, isFetchingNextPage } =
+    usePaginatedBlockList();
+
+  const initialBlockHashes = useMemo(() => {
+    return new Set(initialBlockList.map(block => block.hash));
+  }, [initialBlockList]);
+
+  const initialBurnBlockHashes = useMemo(() => {
+    return new Set(Object.keys(initialBurnBlocks));
+  }, [initialBurnBlocks]);
+
+  const { latestUIBlocks, latestBlocksCount, clearLatestBlocks } = useBlockListWebSocket(
+    initialBlockHashes,
+    initialBurnBlockHashes
+  );
+
+  const showLatestBlocks = useCallback(() => {
+    setLatestBlocksToShow(prevLatestBlocksToShow => {
+      return [...latestUIBlocks, ...prevLatestBlocksToShow];
+    });
+    clearLatestBlocks();
+  }, [clearLatestBlocks, latestUIBlocks]);
+
+  const blockList = useMemo(
+    () => [...latestBlocksToShow, ...initialBlockList],
+    [initialBlockList, latestBlocksToShow]
+  );
+
+  const showLatestBlocksWithFadeEffect = useCallback(() => {
+    setFadeEffect(true);
+    setTimeout(() => {
+      showLatestBlocks();
+      setFadeEffect(false);
+    }, FADE_DURATION);
+  }, [setFadeEffect, showLatestBlocks]);
+
+  const prevLiveUpdatesRef = useRef(liveUpdates);
```

Comment:
You have a usePrevious hook that could be used here

Reply:
This still hasn't been addresssed

Reply:
are you referring to usePrevious in debug.ts?

Reply:
yes

Reply:
I feel using a function from debug in actual logic can cause confusion, I'm planning to remove debug.ts for now and we can discuss if we need to bring it back in a better way.